### PR TITLE
Fix old X-Authorization token prefill logic with session based one

### PR DESCRIPTION
**Pull request by matti-lamppu**
## 🛠️ Changelog
- Fetch token for open city profile prefill from session rather than X-Authorization

## 🎫 Tickets
*This pull request resolves all or part of the following ticket(s):*
- TILA-####

### CODE REVIEW

File: api/graphql/types/reservations/serializers/create_serializers.py [hunk 1]
```diff
@@ -191,7 +192,10 @@ def validate(self, data, prefill_from_profile=True):
         reservation_unit_ids = [x.pk for x in reservation_units]
         self.check_reservation_type(user, reservation_unit_ids, reservation_type)
 
-        if settings.PREFILL_RESERVATION_WITH_PROFILE_DATA and prefill_from_profile:
+        prefill_from_profile = prefill_from_profile and settings.PREFILL_RESERVATION_WITH_PROFILE_DATA
+        id_token = get_id_token(user) or {}
+
+        if prefill_from_profile and is_ad_login(id_token):
```
Comment (vergama) [inline api/graphql/types/reservations/serializers/create_serializers.py:198]: Negation is missing, should attempt the prefill only for non-AD users

Comment (matti-lamppu) [reply api/graphql/types/reservations/serializers/create_serializers.py]: Had it the wrong way around. Fixed.

File: tests/test_graphql_api/test_reservation/test_create.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        "begin": datetime(2023, 1, 1, hour=12).isoformat(),
+        "end": datetime(2023, 1, 1, hour=13).isoformat(),
+        "reservationUnitPks": [reservation_unit.pk],
+    }
+    with mock_profile_reader(), mock_opening_hours(reservation_unit.uuid, date=date(2023, 1, 1)):
+        response = graphql(CREATE_MUTATION, input_data=input_data)
+
+    # then:
+    # - There are no errors in the response
+    # - The reservation is prefilled from the users profile data
```
Comment (vergama) [inline tests/test_graphql_api/test_reservation/test_create.py:122]: Negation also missing from this comment, otherwise ok.

Comment (matti-lamppu) [reply tests/test_graphql_api/test_reservation/test_create.py]: Fixed

**Review by vergama — APPROVED** [review]

**Pull request merged**